### PR TITLE
 🚀 Initial pipeline design

### PR DESCRIPTION
**Pull request by ashwinvaidya17**
## 📝 Description

- Add basic benchmarking
image

```bash
python tools/runners/benchmark.py --config tools/runners/sample.yaml
```

You can also play around with

```bash
python tools/runners/benchmark.py --print_config
```

Addresses: https://github.com/openvinotoolkit/anomalib/issues/1732

## ✨ Changes

Select what type of change your PR is:

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🔨 Refactor (non-breaking change which refactors the code base)
- [x] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation update
- [ ] 🔒 Security update

## ✅ Checklist

Before you submit your pull request, please make sure you have completed the following steps:

- [ ] 📋 I have summarized my changes in the [CHANGELOG](https://github.com/openvinotoolkit/anomalib/blob/main/CHANGELOG.md) and followed the guidelines for my type of change (skip for minor changes, documentation updates, and test enhancements).
- [ ] 📚 I have made the necessary updates to the documentation (if applicable).
- [ ] 🧪 I have written tests that support my changes and prove that my fix is effective or my feature works (if applicable).

For more information about code review checklists, see the [Code Review Checklist](https://github.com/openvinotoolkit/anomalib/blob/main/docs/source/markdown/guides/developer/code_review_checklist.md).

### CODE REVIEW

File: src/anomalib/pipelines/jobs/benchmark.py [hunk 1]
```diff
@@ -0,0 +1,283 @@
+"""Benchmarking job."""
```
Comment (ashwinvaidya17) [inline src/anomalib/pipelines/jobs/benchmark.py:1]: I plan to split this into two files

File: src/anomalib/pipelines/jobs/base.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    @abstractmethod
+    def add_arguments(self, parser: ArgumentParser) -> None:
+        """Add arguments to the parser."""
+
+    @abstractmethod
+    def config_iterator(self, args: Namespace) -> Iterator:
+        """Return iterator based on the arguments."""
+
+
+class BaseJob:
```
Comment (ashwinvaidya17) [inline src/anomalib/pipelines/jobs/base.py:29]: Should inherit from ABC

Comment (samet-akcay) [reply src/anomalib/pipelines/jobs/base.py:29]: And based on our latest discussion, should be Job :)

File: src/anomalib/pipelines/executors/pool.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    from concurrent.futures import Future
+
+logger = logging.getLogger(__name__)
+
+
+class PoolExecutionError(Exception):
+    """Pool execution error should be raised when one or more jobs fail in the pool."""
+
+
+class PoolExecutor(SerialExecutor):
```
[the remark this answers is not in the document]
Comment (danylo-boiko) [reply src/anomalib/pipelines/executors/pool.py:29]: For me, `PoolExecutor` is a more generic name. We can find similar namings in the `concurrent.futures` python module. In my opinion, following such an unspoken agreement will help people who are looking at the anomalib codebase for the first time.

**Pull request closed without merging**